src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -548,6 +548,13 @@ Exception ProcessException(Exception e, ref bool canceled, bool forceWrapExcepti
 
                     canceled = e is OperationCanceledException;
 
+                    // If the operation was canceled, we should not report an error
+                    if (canceled)
+                    {
+                        Abort();
+                        return null!; // Return null but ensure the compiler is happy with the non-nullable return type
```

Comment:
Move the Abort() call above the if statement (and remove the Abort invocation below).

Instead of using `null!`, change the nullability annotations to allow this function to return null.